server-spi/src/main/java/org/keycloak/models/UserModel.java

```diff
@@ -54,6 +54,7 @@ public static class SearchableFields {
         public static final SearchableModelField<UserModel> ID              = new SearchableModelField<>("id", String.class);
         public static final SearchableModelField<UserModel> REALM_ID        = new SearchableModelField<>("realmId", String.class);
         public static final SearchableModelField<UserModel> USERNAME        = new SearchableModelField<>("username", String.class);
+        public static final SearchableModelField<UserModel> USERNAME_CASE_INSENSITIVE = new SearchableModelField<>("usernameCaseInsensitive", String.class);
```

Comment:
These two searchable fields (`USERNAME`, `USERNAME_CASE_INSENSITIVE`) deserve javadoc which states which one is used for case (in)sensitive search